src/run.ts

```diff
@@ -0,0 +1,343 @@
+
+    const clue = summedGameClue(guess, secret);
+
+    console.log(`Clue is: ${clue}`);
+
+    if (clue === 8) {
+      console.log('Congratulations! You guessed the secret!');
+      rl.close();
+    } else {
+      // Continue asking for the next guess
+      summedGame();
+    }
+  });
+};
+
+/**
+ * A function to simulate the game with classical mastermind game clues in the console.
+ */
+const classicalGame = () => {
+  rl.question('Enter a guess (4 digits): ', (answer) => {
+    if (!/^\d{4}$/.test(answer)) {
+      console.log('You should enter exactly 4 digits!');
+      currentGame();
+      return;
+    }
+    const guess = answer.split('').map(Number);
+
+    console.log(`You entered: '${answer}'`);
+
+    const clue = classicalGameClue(guess, secret);
+
+    console.log(`Clue : Hit: ${clue.hit} Blow: ${clue.blow}`);
+    if (clue.hit === 4) {
+      console.log('Congratulations! You guessed the secret!');
+      rl.close();
+    } else {
+      // Continue asking for the next guess
+      classicalGame();
+    }
+  });
+};
+
+/**
+ * A function to simulate the game with shuffled clues in the console.
+ */
+const shuffledClueGame = () => {
+  rl.question('Enter a guess (4 digits): ', (answer) => {
+    if (!/^\d{4}$/.test(answer)) {
+      console.log('You should enter exactly 4 digits!');
+      currentGame();
+      return;
+    }
+
+    const guess = answer.split('').map(Number);
+
+    console.log(`You entered: '${answer}'`);
+
+    const clue = shuffledGameClue(guess, secret);
+
+    console.log(`Clue is: ${clue.join(' ')}`);
+
+    // If all positions match, end the game
+    if (clue.join('') === '2222') {
+      console.log('Congratulations! You guessed the secret!');
+      rl.close();
+    } else {
+      // Continue asking for the next guess
+      shuffledClueGame();
+    }
+  });
+};
+
+/**
+ * A function to simulate the game with shifted clues in the console.
+ */
+const shiftedClueGame = () => {
+  rl.question('Enter a guess (4 digits): ', (answer) => {
+    if (!/^\d{4}$/.test(answer)) {
+      console.log('You should enter exactly 4 digits!');
+      currentGame();
+      return;
+    }
+    const guess = answer.split('').map(Number);
+
+    console.log(`You entered: '${answer}'`);
+
+    const clue = shiftedGameClue(guess, secret);
+
+    console.log(`Clue is: ${clue.join(' ')}`);
+    // If all positions match, end the game
+    if (clue.join('') === '2222') {
+      console.log('Congratulations! You guessed the secret!');
+      rl.close();
+    } else {
+      // Continue asking for the next guess
+      shiftedClueGame();
+    }
+  });
+};
+
+/**
+ * A function to simulate the game with current clue system in the console.
+ */
+const currentGame = () => {
+  rl.question('Enter a guess (4 digits): ', (answer) => {
+    if (!/^\d{4}$/.test(answer)) {
+      console.log('You should enter exactly 4 digits!');
+      currentGame();
+      return;
+    }
+
+    const guess = answer.split('').map(Number);
+
```

Comment:
```suggestion
    if (new Set(guess).size !== 4 || guess.some((num) => num < 1 || num > 9)) {
      console.log('Digits should be unique and between 1-9!');
      currentGame();
      return;
    }
```
We can check for digits to be unique and between 1-9 here